### ISSUE TITLE
[Discussion] Linting recipe in chapter 6?

### ISSUE THREAD

**Issue by bast** [bast/cmake-recipes#166]
We could add a recipe in chapter 6 to do some linting of source code at build time - for instance not compile if something obviously terrible is being done. Something like detecting implicit.h. Basically what this example would be about is run a python script at build time and stop compilation if some predicate is met or not met.

Perhaps this is bad idea but wanted to write it down before I forget.

**Comment by robertodr** [comment]
I don't think it's a bad idea.

**Comment by robertodr** [comment]
Probably something that runs `clang-format` before building (annoying in the real world) or `clang-tidy`? Or `clang-analyzer`? https://github.com/ttroy50/cmake-examples/tree/master/04-static-analysis/clang-analyzer

**Comment by bast** [comment]
Thanks! I was thinking of some example that compilers cannot do, that is relevant, better done at compile time than as git hook, and not annoying. I will continue brainstorming ...

**Comment by bast** [comment]
The more I think about it, the more I am convinced that this should be decoupled from compilation. I know of one use case where somebody wanted this as a sanity check and I have implemented this (in Dalton) but I think this might not be a general-interest recipe and perhaps does not warrant a place among the top 100 recipes. I think in most cases this can be given to a compiler or an another tool or a Git hook or CI. Closing.
